src/components/workflow/workflow.service.ts

```diff
@@ -0,0 +1,1033 @@
+              id: input.stateId
+            }),
+            relation('out', 'rel', 'notification'),
+            node('sg')
+          ]
+        ])
+        .detachDelete('rel')
+        .first();
+
+    } catch (e) {
+      this.logger.warning('could not remove security group from state', {
+        exception: e
+      });
+      throw e;
+    }
+  }
+
+  // changeCurrentStateInWorkflow
+  async changeCurrentState(session: ISession, input: ChangeState): Promise<void>{
+    try{
+      const currentStateAndWorkflow = await this.db
+        .query()
+        .match([
+          [
+            ...matchSession(session),
+            relation('in', '', 'member', {
+              admin: true
+            }),
+            node('sg', 'SecurityGroup'),
+            relation('out', '', 'permission', {
+              read: true
+            }),
+            node('permission', 'Permission', {
+              canRead: true
+            }),
+            relation('out', '', 'baseNode'),
+            node('baseNode', 'BaseNode'),
+            relation('out', '', 'currentState'),
+            node('currentState')
+          ],
+          [
+            node('baseNode'),
+            relation('out', '', 'workflow', {
+              active: true
+            }),
+            node('workflow')
+          ]
+        ])
+        .return({
+          currentState: [
+            { value: 'value' },
+          ],
+          workflow: [
+            { stateIdentifier: 'stateIdentifier' }
+          ]
+        })
+        .first();
+
+      if( !currentStateAndWorkflow ) {
+        throw new NotFoundException('could not find current state and workflow');
+      }
+
+      const possibleState = await this.db
+        .query()
+        .match([
+          [
+            ...matchSession(session),
+            relation('in', '', 'member', {
+              admin: true
+            }),
+            node('sg', 'SecurityGroup'),
+            relation('in', '', 'securityGroup', {
+              active: true
+            }),
+            node('state', 'State', {
+              value: currentStateAndWorkflow.value
+            }),
+            relation('out', '', 'nextPossibleState', {
+              active: true
+            }),
+            node('state', 'State', {
+              id: input.newStateId
+            })
+          ]
+        ])
+        .return({
+          state: 'state'
+        })
+        .first();
+
+      if( !possibleState ) {
+        throw new NotFoundException('new state provided is not a nextpossiblstate of current state');
+      }
+
+      await this.db
+        .query()
+        .match([
+          [
+            ...matchSession(session),
+            relation('in', '', 'member', {
+              admin: true
+            }),
+            node('sg', 'SecurityGroup'),
+            relation('out', '', 'permission', {
+              write: true
+            }),
+            node('permission', 'Permission', {
+              canWrite: true,
```

Comment:
I was wrong, the schema shows the property conferring write permission is called `write`